app/seeders/admin_user_seeder.rb

```diff
@@ -26,24 +26,30 @@
 #
 # See doc/COPYRIGHT.rdoc for more details.
 #++
+class AdminUserSeeder
 
-if ReportedProjectStatus.any?
-  puts '***** Skipping reported project status as there are already some configured'
-else
-  ReportedProjectStatus.transaction do
-    ReportedProjectStatus.new.tap do |status|
-      status.name = I18n.t(:default_reported_project_status_green)
-      status.is_default = true
-    end.save!
+  def self.seed!
+    if User.admin.empty?
+      user = User.new
```

Comment:
If put everything into methods this offers opportunities for plugins to amend things. For example the mobile_top plugin. With that plugin seeding the admin will fail as the user doesn't have a mobile number. Let's change this code to something like this:

```ruby
class AdminUserSeeder
  def seed
    if User.admin.empty?
      user = new_admin
      user.save!
    end
  end

  def new_admin
    User.new.tap do |u|
      user.admin = true
      user.login = 'admin'
      user.password = '!AdminAdminAdmin123%&/'
      user.firstname = 'OpenProject'
      user.lastname = 'Admin'
      user.mail = ENV.fetch('ADMIN_EMAIL') { 'admin@example.net' }
      user.mail_notification = User::USER_MAIL_OPTION_NON.first
      user.language = I18n.locale.to_s
      user.status = User::STATUSES[:active]
    end
  end
end
```

Then in the mobile_otp plugin we can work some magic and amend the mobile number:

```ruby
module AdminWithMobileNumber
  def new_admin
    super.tap do |admin|
      admin.mobile_number = '+49 123456' # no idea how the actual method is called but you get the idea
    end
end
```

What do you think @leewaa @oliverguenther @ulferts ?